icu4j/main/classes/core/src/com/ibm/icu/util/ULocale.java

```diff
@@ -1446,10 +1508,21 @@ private boolean replaceLanguage(boolean checkLanguage,
                         this.variants.set(variantIndex, replacedVariant);
                     } else {
                         this.variants.remove(variantIndex);
+                        if (this.variants.isEmpty()) {
+                            this.variants = null;
+                        }
                     }
                 }
-                if (replacedExtensions != null) {
-                    this.extensions = replacedExtensions;
+                if (replacedExtensions != null && !replacedExtensions.isEmpty()) {
+                    // DO NOTHING
+                    // UTS35 does not specifiy what should we do if we have extensions in the
+                    // replacement. Currently we know only the following 4 "BCP47 LegacyRules" have
+                    // extensions in them:
+                    //  i_default => en_x_i_default
+                    //  i_enochian => und_x_i_enochian
+                    //  i_mingo => see_x_i_mingo
+                    //  zh_min => nan_x_zh_min
+                    // But all of them are already change my LanguageTag before hitting this code.
```

Comment:
I don't know what this line means. Please clarify.

Reply:
Code in LanguageTag ( see https://github.com/unicode-org/icu/blob/9971c663ff89e1bf78f9c4805daa407af8a023cb/icu4j/main/classes/core/src/com/ibm/icu/impl/locale/LanguageTag.java#L130 ) would be run before calling into this replace function. So the code inside the replace function will never see these legacy tags.

Reply:
Ok, so it's just typos. --> But all of them are already changed by class LanguageTag before hitting this code.